packages/console/src/hooks/use-api-resources-usage.ts

```diff
@@ -0,0 +1,37 @@
+import { isManagementApi } from '@logto/schemas';
+import { useContext, useMemo } from 'react';
+import useSWR from 'swr';
+
+import { type ApiResource } from '@/consts';
+import { isCloud } from '@/consts/env';
+import { TenantsContext } from '@/contexts/TenantsProvider';
+import { hasReachedQuotaLimit } from '@/utils/quota';
+
+import useSubscriptionPlan from './use-subscription-plan';
+
+const useApiResourcesUsage = () => {
+  const { currentTenantId } = useContext(TenantsContext);
+  const { data: currentPlan } = useSubscriptionPlan(currentTenantId);
+  /**
+   * Note: we only need to fetch all resources when the user is in cloud environment.
+   * The oss version doesn't have the quota limit.
+   */
+  const { data: allResources } = useSWR<ApiResource[]>(isCloud && 'api/resources');
```

Comment:
How about a hook `useUsage(key)`, and return `hasReachedLimit` false for non cloud environment

Reply:
When more usage hooks are introduced in the future, we will consider implementing a unified approach for handling them. For now, we are adopting a more straightforward method for better comprehension.